File: src/ve/org/bcv/services/Encuesta_caService.java
package ve.org.bcv.services;

import java.util.Date;
import java.util.List;

import javax.ejb.Local;

import ve.org.bcv.dto.BenefMedicamentoDto;
import ve.org.bcv.dto.ChartDto;
import ve.org.bcv.dto.ParametroNominaDto;
import ve.org.bcv.dto.ResultBoleanDto;
import ve.org.bcv.dto.UserEncuestaDto;

@Local
public interface Encuesta_caService {
	List<ParametroNominaDto> findByCoProcesoNom(String co_proceso_nom);
	boolean save(BenefMedicamentoDto benefMedicamentoDto,List<ParametroNominaDto> lista);
	UserEncuestaDto findUserEncuesta(String cedula);
	ResultBoleanDto personaYaEncuestada( String cedula,String nuCedulaFamiliar);
	ChartDto graficar(String interesdos, String feDesdeStr, String feHastaStr);
	List<ParametroNominaDto> findByCoProcesoNomNo(String co_proceso_nom,String tx_parametro_nom) ;
	List<ParametroNominaDto> buscarMedicamento(String filtro) ;
	UserEncuestaDto buscarCedulaMedicamentos(String cedula) ;
	UserEncuestaDto buscarCedulaMedicamentos(String cedula,String cedulaFamiliar) ;
	 List<ParametroNominaDto>  buscarMedicamentosFamiliar(String cedula,String cedulaFamiliar) ;
	 List<UserEncuestaDto> buscarCedulaMedicamentos(Date feDesde,Date feHasta);
	List<String>  buscarCedulaByFechas(Date feDesde, Date feHasta) ;
	 List<UserEncuestaDto> buscarCedulaMedicamentosQuery (String feDesdeStr, String fechaHastaStr);
	 List<UserEncuestaDto> familiares(String cedula) ;
	
}

File: WebContent/js/service/encuesta_caService.js
'use strict';

app.service('encuesta_caService',
    function encuesta_caService($resource) {
        return $resource('services/encuesta_caService/:co_proceso_nom', {
        	co_proceso_nom: '@co_proceso_nom'
        }, {
            query: {
                method: 'GET',
                isArray: false
            },
            buscarMedicamento: {
            	url:'services/encuesta_caService/buscarMedicamento/:co_proceso_nom',
                method: 'GET',
                isArray: false
            },
            
            buscarCedulaMedicamento: {
            	url:'services/encuesta_caService/buscarCedulaMedicamento/:cedula',
                method: 'GET',
                isArray: false
            },
            buscarCedulaMedicamentos: {
            	url:'services/encuesta_caService/buscarCedulaMedicamentos/:feDesde/:feHasta',
                method: 'GET',
                isArray: true
            },
            buscarCedula: {
            	url:'services/encuesta_caService/buscarCedula/:co_proceso_nom',
                method: 'GET',
                isArray: false
            },
            queryNo: {
            	url:'services/encuesta_caService/listNo/:co_proceso_nom/:tx_parametro_nom',
                method: 'GET',
                isArray: false
            },
            findUser: {
            	url:'services/encuesta_caService/findUser/:cedula',
                method: 'GET',
                isArray: false
            },
            parametroNominaDtos : {
            	url:'services/encuesta_caService/parametroNominaDtos/:cedula/:nuCedulaFamiliar',
                method: 'GET',
                isArray: true
            }, 
            familiares : {
            	url:'services/encuesta_caService/familiares/:cedula',
                method: 'GET',
                isArray: true
            },
            
            
            graficar: {
            	url:'services/encuesta_caService/graficar/:interesdos/:feDesde/:feHasta',
                method: 'GET',
                isArray: false
            }  
        });
    });


 